src/NzbDrone.Core/Download/Clients/rTorrent/RTorrent.cs

```diff
@@ -0,0 +1,217 @@
+                // Remove from client, since it is discarded
+                RemoveItem(hash, false);
+
+                return null;
+            }
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {
+            _proxy.AddTorrentFromFile(filename, fileContent, Settings);
+
+            _proxy.SetTorrentLabel(hash, Settings.TvCategory, Settings);
+
+            var priority = (RTorrentPriority)(remoteEpisode.IsRecentEpisode() ?
+                                    Settings.RecentTvPriority : Settings.OlderTvPriority);
+            _proxy.SetTorrentPriority(hash, Settings, priority);
+
+            return hash;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "rTorrent";
+            }
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            try
+            {
+                var torrents = _proxy.GetTorrents(Settings);
+
+                _logger.Debug("Retrieved metadata of {0} torrents in client", torrents.Count);
+
+                var items = new List<DownloadClientItem>();
+                foreach (RTorrentTorrent torrent in torrents)
+                {
+                    // Don't concern ourselves with categories other than specified
+                    if (torrent.Category != Settings.TvCategory) continue;
+
+                    if (torrent.Path.StartsWith("."))
+                    {
+                        throw new DownloadClientException("Download paths paths must be absolute. Please specify variable \"directory\" in rTorrent.");
+                    }
+
+                    var item = new DownloadClientItem();
+                    item.DownloadClient = Definition.Name;
+                    item.Title = torrent.Name;
+                    item.DownloadId = torrent.Hash;
+                    item.OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(torrent.Path));
+                    item.TotalSize = torrent.TotalSize;
+                    item.RemainingSize = torrent.RemainingSize;
+                    item.Category = torrent.Category;
+
+                    if (torrent.DownRate > 0) {
+                        var secondsLeft = torrent.RemainingSize / torrent.DownRate;
+                        item.RemainingTime = TimeSpan.FromSeconds(secondsLeft);
+                    }
+
+                    if (torrent.IsFinished) item.Status = DownloadItemStatus.Completed;
+                    else if (torrent.IsActive) item.Status = DownloadItemStatus.Downloading;
+                    else if (!torrent.IsActive) item.Status = DownloadItemStatus.Paused;
+                    else if (!torrent.IsOpen) item.Status = DownloadItemStatus.Queued;
+
+                    // Since we do not know the user's intent, do not let the Sonarr to remove torrent
+                    item.IsReadOnly = true;
```

Comment:
Hm, I don't know if rTorrent pauses/stops torrents if seeding slots are fully used. So just deleting it isn't a good idea, with risk of hnr and all.
For Deluge I actually have some compare logic that compares the ratio with deluge own config. Mainly because Deluge would 'pause' torrents that finished seeding, and doesn't have a stopped state.

Dunno how that it for rTorrent. This is pretty much why we need tests. So we know how rTorrent behaves.